### ISSUE TITLE
Abstract <ViroARScene /> out of main file using ES6 format

### ISSUE THREAD

**Issue by dittmarconsulting** [viromedia/viro#60]
Hi Andi,

I'm aware that your code is still using the ES5 format but maybe you can help me with the my implementation.

I have reproduced your implementation as follows:

**Main.js**
```
const ARScenes = require('../components/ARScenes')
...
<ViroARSceneNavigator
    apiKey='<key>'
    initialScene={{scene: ARScenes}} />
```

**ARScenes.js**
```
const ARScenes = React.createClass({
  render: function() {
    return (
      <ViroARScene>
      </ViroARScene>
    )
  }
})
module.exports = ARScenes
```

Now that's done with the old format.

When I try to use ES6 as follows:

**Main.js**
```
import ARScenes from '../components/ARScenes'
...
<ViroARSceneNavigator
    apiKey='<key>'
    initialScene={{scene: ARScenes}} />
```
**ARScenes.js**
```
class ARScenes extends Component {
  render() {
    return (
      <ViroARScene>
      </ViroARScene>
    )
  }
}
export default ARScenes
```

it works as well but once I have child components inside `<ViroARScene>` like that:

```
class ARScenes extends Component {
  render() {
    return (
      <ViroARScene>
           <ARPlaneSelector>
               ... other components
           </ARPlaneSelector>
      </ViroARScene>
    )
  }
}
```

I get this error: 
`React.createElement: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: undefined. You likely forgot to export your component from the file it's defined in.`

If I have everything in the **Main.js** it works without any issues.

Any idea how I can mitigate that @achuvm ?

**Comment by achuvm** [comment]
Hi @dittmarconsulting,

I don't see why that wouldn't work since we've written samples using ES6 classes (even with our ES5/React.createClass components).

While we've personally haven't seen these issues:

1. Did you export the ARScenes? Do you have the line `export default ARScenes` at the bottom of the new ES6 class file?
2. `ARPlaneSelector` should be `ViroARPlaneSelector` although, that could just be the name of your implementation of it (I remember you modified your own copy).

**Comment by dittmarconsulting** [comment]
Hi Andy,

Yes, `ARPlaneSelector` is my own implementation and I have `export default ARScenes` at the bottom of my component like that

```
class ARScenes extends Component {
  render() {
    return (
      <ViroARScene>
           <ARPlaneSelector>
               ... other components
           </ARPlaneSelector>
      </ViroARScene>
    )
  }
}
export default ARScenes
```

As soon I delete the children of the `ARScenes` it works. I'm wondering whether there is a "racing" issue going on. What do you think?

**Comment by achuvm** [comment]
Hi @dittmarconsulting,

What do you mean by children of the `ARScenes`?

In your example:
```
return (
  <ViroARScene>
    <ARPlaneSelector>
      ... other components
    </ARPlaneSelector>
  </ViroARScene>
)
```

Is `<ARPlaneSelector>` the child or are `... other components` the children components causing the issue?

I don't think there's a race condition here, but I am wondering if it's some forgotten/malformed `import` statement.

**Comment by dittmarconsulting** [comment]
Hi Andy,

Just for clarification. If I have no children (components within `< ViroARScene />`) like that:

```
class ARScenes extends Component {
  render() {
    return (
      <ViroARScene>
      </ViroARScene>
    )
  }
}
export default ARScenes
```

it works but once I have child components (even just `<ARPlaneSelector />`) like that

```
class ARScenes extends Component {
  render() {
    return (
      <ViroARScene>
           <ARPlaneSelector />
      </ViroARScene>
    )
  }
}
export default ARScenes
```

I will get the error.

**Comment by achuvm** [comment]
Hi @dittmarconsulting,

I just created a new Viro project and my `index.ios.js` looks like the following:

```
import React, { Component } from 'react';
import {
  AppRegistry,
} from 'react-native';

import {
  ViroARSceneNavigator,
} from 'react-viro';

import ARScenes from './js/ARScenes';

export default class ViroSample extends Component {
  render() {
    return (
      <ViroARSceneNavigator apiKey="YOUR_API_KEY_HERE"
        initialScene={{scene: ARScenes}}
        />
    );
  }
}

// The below line is necessary for use with the TestBed App
AppRegistry.registerComponent('ViroSample', () => ViroSample);
```

My `./js/ARScenes.js` file looks like this:

```
'use strict';

import React, { Component } from 'react';

import {
  ViroARScene,
  ViroText,
  ViroARPlaneSelector,
} from 'react-viro';


class ARScenes extends Component {
  render() {
    return (
      <ViroARScene>
        <ViroARPlaneSelector />
      </ViroARScene>
    )
  }
}
export default ARScenes
```

Can you try these and let me know if they work? I tried this with the latest Viro Media App and got it to work. I think it should be set up like yours.

Thanks

Andy

**Comment by dittmarconsulting** [comment]
That's exactly how I had it. Let me try again and investigate.

Thanks,
Andy

**Comment by dittmarconsulting** [comment]
DOH! Far out, I had a spelling mistake in my code.

Sorry for the time wasted :-(

Cheers,
Thomas

**Comment by achuvm** [comment]
No problem!

Glad you got it working!

Andy
